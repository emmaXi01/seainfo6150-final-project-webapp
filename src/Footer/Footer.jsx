import React from "react";
import LinkList from '../LinksList/LinksList.jsx';
import styles from "./Footer.module.css";

const Footer = () => {
  const basicInfo = ["About", "Leadership", "Careers", "Catalog"];
  const community = ["Learners", "Blog", "Tech Blog"];
  const more = ["Terms", "Privacy", "Help", "Contact"];

  return (
    <div className={styles.footerBlock}>
      <div className={styles.listContainer}>
        <div className={styles.title}>OnlineEdu</div>
        <LinkList links={basicInfo}/>
      </div>
      <div className={styles.listContainer}>
        <div className={styles.title}>Community</div>
        <LinkList links={community}/>
      </div>    
      <div className={styles.listContainer}>
        <div className={styles.title}>More</div>
        <LinkList links={more}/>
      </div>
      <div className={styles.listContainer}>
        <div className={styles.title}>Others</div>
      </div>      
    </div>
  );
};

export default Footer;