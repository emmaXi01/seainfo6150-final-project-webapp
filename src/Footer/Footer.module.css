.footerBlock {
  display: grid;
  margin: 0px 12px;
  padding: 20px;
}

.listContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  position: relative;
}

.title {
  font-weight: 600;
}

@media screen and (min-width: 400px) {
  .footerBlock {
    grid-template-columns: repeat(2, 1fr);
  }
} 


@media screen and (min-width: 500px) {
  .footerBlock {
    grid-template-columns: repeat(3, 1fr);
  }
}

@media screen and (min-width: 700px) {
  .footerBlock{
    grid-template-columns: repeat(4, 1fr);
  }
}