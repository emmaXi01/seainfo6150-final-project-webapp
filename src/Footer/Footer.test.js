import React from "react";
import { render } from "@testing-library/react";
import Footer from "./Footer.jsx";

describe("Footer tests", () => {
  it("renders the Footer component correctly", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});