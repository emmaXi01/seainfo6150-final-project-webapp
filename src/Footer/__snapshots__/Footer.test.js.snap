// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer tests renders the Footer component correctly 1`] = `
<div>
  <div
    class="footerBlock"
  >
    <div
      class="listContainer"
    >
      <div
        class="title"
      >
        OnlineEdu
      </div>
      <div>
        <ul
          class="linksList"
        >
          <li>
            About
          </li>
          <li>
            Leadership
          </li>
          <li>
            Careers
          </li>
          <li>
            Catalog
          </li>
        </ul>
      </div>
    </div>
    <div
      class="listContainer"
    >
      <div
        class="title"
      >
        Community
      </div>
      <div>
        <ul
          class="linksList"
        >
          <li>
            Learners
          </li>
          <li>
            Blog
          </li>
          <li>
            Tech Blog
          </li>
        </ul>
      </div>
    </div>
    <div
      class="listContainer"
    >
      <div
        class="title"
      >
        More
      </div>
      <div>
        <ul
          class="linksList"
        >
          <li>
            Terms
          </li>
          <li>
            Privacy
          </li>
          <li>
            Help
          </li>
          <li>
            Contact
          </li>
        </ul>
      </div>
    </div>
    <div
      class="listContainer"
    >
      <div
        class="title"
      >
        Others
      </div>
    </div>
  </div>
</div>
`;
